File: EliteFramework/UIColor/EliteColor.m
//
//  EliteCategories.m
//  BLEDrawingPad
//
//  Created by Lifelong-Study on 2015/4/21.
//  Copyright (c) 2015年 PaoYo Ding. All rights reserved.
//

#import "EliteColor.h"

@implementation UIColor (EliteColorCategory)

+ (UIColor *)colorWithHex:(unsigned int)hex
{
    return [self colorWithHex:hex alpha:1];
}

+ (UIColor *)colorWithHex:(unsigned int)hex alpha:(CGFloat)alpha
{
    return [UIColor colorWithRed:((float)((hex & 0xFF0000) >> 16)) / 255.0
                           green:((float)((hex & 0xFF00) >> 8)) / 255.0
                            blue:((float)(hex & 0xFF)) / 255.0
                           alpha:alpha];
}

+ (NSInteger)getRedColorWithHex:(unsigned int)hex
{
    return (hex & 0xFF0000) >> 16;
}

+ (NSInteger)getGreenColorWithHex:(unsigned int)hex
{
    return (hex & 0x00FF00) >> 8;
}

+ (NSInteger)getBlueColorWithHex:(unsigned int)hex
{
    return hex & 0x00FF;
}

@end

File: EliteFramework/EliteColor.swift
//
//  EliteColor.swift
//  EliteFramework
//
//  Created by Lifelong-Study on 2015/12/17.
//  Copyright © 2015年 Lifelong-Study. All rights reserved.
//

import Foundation
import UIKit

public extension UIColor {
    convenience init(hex: UInt64) {
        self.init(  red: ((CGFloat)((hex & 0xFF0000) >> 16)) / 255.0,
                    green: ((CGFloat)((hex & 0xFF00) >> 8)) / 255.0,
                    blue: ((CGFloat) (hex & 0xFF)) / 255.0,
                    alpha: 1.0)
    }
    
    convenience init(hex: UInt64, alpha: CGFloat) {
        self.init(  red: ((CGFloat)((hex & 0xFF0000) >> 16)) / 255.0,
                    green: ((CGFloat)((hex & 0xFF00) >> 8)) / 255.0,
                    blue: ((CGFloat) (hex & 0xFF)) / 255.0,
                    alpha: alpha)
    }
    
    convenience init(_ red: Int, _ green: Int, _ blue: Int) {
        self.init(red, green, blue, alpha: 1.0)
    }
    
    convenience init(_ red: Int, _ green: Int, _ blue: Int, alpha: CGFloat) {
        self.init(red: CGFloat(red) / 255.0, green: CGFloat(green) / 255.0, blue: CGFloat(blue) / 255.0, alpha: alpha)
    }
    
    static let random = UIColor(red: CGFloat.random(in: 0...1), green: CGFloat.random(in: 0...1), blue: CGFloat.random(in: 0...1), alpha: 1.0)
}
